### DISCUSSION TITLE
Stream response gets truncated in vercel deployment

### DISCUSSION THREAD

**Discussion by brenorosado** [vercel/next.js#69800] [Help] [answered]
### Summary

When returning a stream from route like:
```
return new NextResponse(stream, {
    headers: {
      "Content-Type": "text/plain",
      "Transfer-Encoding": "chunked",
      "x-no-compression": "1",
      "Cache-Control": "no-store"
    },
  });
```

  
The content is loaded just fine locally. But when the app is deployed to vercel, the stream response always gets truncated.

What I got so far is that the "Transfer-Enconding" header is not returned in the response header when deployed to vercel.
Does anyone know how to fix this?

**Comment by Shriharsh-Deshmukh** [comment]
Hey **Breno!** 
Vercel's serverless functions sometimes handle streams differently, which could explain why the "Transfer-Encoding" header is missing and the response gets truncated. 

Here are a few things you could try:
1. **Use Vercel Edge Functions** instead of serverless functions. Edge functions might handle streaming better.
2. **Check the Vercel limits** for stream size or timeout, as your stream might be getting cut off due to platform constraints.
3. **Add an explicit `Content-Length` header** if the size of the response is known, which can help avoid truncation.

**Comment by virattt** [comment]
Having the same issue.  I am using the default AI Chatbot project that Vercel has built: https://github.com/vercel/ai-chatbot/tree/main

**Reply by brenorosado** [reply, marked as answer]
That issue occurred in my app due to the 'Function Max Duration' setting in the Vercel project. The default maximum duration for serverless functions is 10 seconds, but it can be increased to 60 seconds on the free tier.

**Reply by virattt** [reply]
@brenorosado - where did you increase the duration?

Thanks

**Reply by brenorosado** [reply]
Vercel project panel > Settings >  Functions > Function Max Duration

**Comment by hoqua** [comment]
Been hitting the 60s timeout on Vercel when streaming longer responses, so I built https://www.durablr.run/ to solve it. It handles the streaming on its side without time limits, and I just collect the output from Vercel as it comes in. Makes long-running tasks way more reliable.
